protocols/smtp/src/main/java/org/apache/james/protocols/smtp/hook/HookReturnCode.java

```diff
@@ -19,15 +19,78 @@
 
 package org.apache.james.protocols.smtp.hook;
 
-import com.google.common.collect.ImmutableList;
+import java.util.Objects;
+
+import com.google.common.base.MoreObjects;
 
 public class HookReturnCode {
-    public static final int OK = 0x1;
-    public static final int DENY = 0x1 << 1;
-    public static final int DENYSOFT = 0x1 << 2;
-    public static final int DECLINED = 0x1 << 3;
-    public static final int DISCONNECT = 0x1 << 4;
-
-    public static final ImmutableList<Integer> VALID_RETURN_CODE = ImmutableList.of(
-        DECLINED, OK, DENY, DENYSOFT, DISCONNECT);
+
+    public enum Action {
+        OK,
+        DENY,
+        DENYSOFT,
+        DECLINED,
+        NONE
+    }
+
+    public enum Disconnection {
+        Disconnect,
+        KeepConnected
+    }
+
+    public static HookReturnCode denySoft() {
+        return new HookReturnCode(Action.DENYSOFT, Disconnection.KeepConnected);
```

Comment:
what about having a `connected` factory method ? (you would then call `return connected(DENYSOFT);`)